js/coinsbit.js

```diff
@@ -0,0 +1,721 @@
+                    // '401': AuthenticationError, // Invalid signature, fail to pass the validation
+                    // '429': RateLimitExceeded, // too many requests, rate limit rule is violated
+                    // '1000': PermissionDenied, // {"success":false,"code":1000,"message":"Please open contract account first!"}
+                    // '1002': InvalidOrder, // {"success":false,"code":1002,"message":"Contract not allow place order!"}
+                    // '10072': AuthenticationError, // Invalid access key
+                    // '10073': AuthenticationError, // Invalid request time
+                    // '10216': InvalidAddress, // {"code":10216,"msg":"No available deposit address"}
+                    // '10232': BadSymbol, // {"code":10232,"msg":"The currency not exist"}
+                    // '30000': BadSymbol, // Trading is suspended for the requested symbol
+                    // '30001': InvalidOrder, // Current trading type (bid or ask) is not allowed
+                    // '30002': InvalidOrder, // Invalid trading amount, smaller than the symbol minimum trading amount
+                    // '30003': InvalidOrder, // Invalid trading amount, greater than the symbol maximum trading amount
+                    // '30004': InsufficientFunds, // Insufficient balance
+                    // '30005': InvalidOrder, // Oversell error
+                    // '30010': InvalidOrder, // Price out of allowed range
+                    // '30016': BadSymbol, // Market is closed
+                    // '30019': InvalidOrder, // Orders count over limit for batch processing
+                    // '30020': BadSymbol, // Restricted symbol, API access is not allowed for the time being
+                    // '30021': BadSymbol, // Invalid symbol
+                    // '33333': BadSymbol, // {"code":33333,"msg":"currency can not be null"}
+                },
+                'broad': {
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const [defaultType, request] = this.handleMarketTypeAndParams ('fetchMarkets', undefined, params); // Note: imho, I am against dividing methods according to types (like 'fetchSpotMarkets', 'fetchFuturesMarkets' and etc..) because there might be many types and exchanges, from time to time, add new types, thus it will need a never-ending 'new method' creations. I think that, independent to the exchange-type, everything should be done with one 'fetchXXX' functions - if there are multiple different types, they should be handled inside one 'fetchXXX' function (just by passing 'defaultType' parameter). That has also one advantage - the most of the 'return' object structure can be same and only a few property-values might need to be happening inside if-else blocks, thus will save multiple 'fetchSpotXXX', 'fetchFuturesXXX' functions to be written over-and-over again with same structure.
+        let data = '';
```

Comment:
We don't need to initialize it to a string, if they always return an array. If the value is unknown at this time, we should initialize to `undefined`.

Reply:
done